apps/sertra/components/PackageDataFetcher.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
```

Comment:
This whole file is in the wrong location, and uses the wrong file extension.

The `components` directory should only be used for react components, and since this is simply typescript code, it should be placed elsewhere. Likewise there's no reason to use the `.tsx` extension (which is for react templates), use `.ts` instead (plain typescript).

What this actually is, is a hook, and it's fairly common to have one or more modules called `hooks` somewhere in a react project, which contain custom hook implementations.